markdown/3.x/en/documentation/single-hit-redirect.md

```diff
@@ -0,0 +1,104 @@
+By definition single hit redirect is a mechanism which redirects user to record page only if he did the exact search i.e. asked for a specific product by providing its identifier.
+
+Built-in mechanism checks the `resultArticleNumberStatus` property of search result object in order to determine if response contains only one record returned with 100% search similarity.
+That level of similarity is only possible to achieve if user search for the product using its number which in most cases is the only one way to determine if exact search takes place. 
+Here is the full condition used in built-in single hit redirect mechanism.
+`result.resultArticleNumberStatus === 'resultsFound' && result.records && result.records.length === 1`
+
+**NOTE** Value `resultFound` is returned only when user searches for a product by passing its number as a search phrase.
+ Product number is the field which has a role `displayProductNumber`.
+ For more information about field roles, see the [FieldRoles](/documentation/3.x/field-roles).
+ Although it is not necessary to check the `result.resultArticleNumberStatus` we highly recommend checking that value to avoid redirecting to the record that is not necessarily the one, user was looking for.
+
+```html
+<script>
+        document.addEventListener("WebComponentsReady", function () {
+            factfinder.communication.ResultDispatcher.subscribe('result', function(result) {
+                if (result.resultArticleNumberStatus === 'resultsFound' 
+                    && result.records 
+                    && result.records.length === 1) {
+                        const record = result.records[0].record;
+                        window.location.href = record.Deeplink; //field name might be different. Check your feed file 
+                }       
+        });
+    });
+</script> 
+```
+**NOTE**  Example above wil not working fully. Please continue with the article to learn more about the single hit redirect insights.
+
+
+### NG compatibility
+For Web Components version lower than `3.15.2` field `resultArticleNumberStatus` is not available when using FACT-Finder NG. Please use `articleNumberSearch` instead.
+Please keep in mind that this field stores boolean values. 
+
+  ```javascript
+ if (result.articleNumberSearch && result.records && result.records.length === 1) {
+    //redirect user
+}      
+```
+
+**NOTE** All other examples in this article use `resultArticleNumberStatus`. 
+
+
+### Preventing redirection after getting back from record page
+If `search-immediate` is enabled, user might find himself in a situation where he will be unable to go back to the search result page, as the single hit redirect will keep redirecting him to record page.
+In order to prevent this some information have to be stored and passed between documents. That information should be intended to inform the single hit redirect mechanism to stop redirecting user as he is just got back from the record page, he was redirected to.
```

Comment:
```suggestion
In order to prevent this some information has to be stored and passed between documents. This information is intended to inform the single hit redirect mechanism to stop redirecting users when they got back from the record page they were just redirected to.
```